pkg/server/container_create.go

```diff
@@ -158,12 +158,23 @@ func (c *criContainerdService) CreateContainer(ctx context.Context, r *runtime.C
 		}
 	}()
 
-	// Update container CreatedAt.
-	meta.CreatedAt = time.Now().UnixNano()
+	container, err := containerstore.NewContainer(meta, containerstore.Status{CreatedAt: time.Now().UnixNano()})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create internal container object for %q: %v",
+			id, err)
+	}
+	defer func() {
+		if retErr != nil {
+			// Cleanup container status checkpoint on error.
+			if err := container.Status.Delete(); err != nil {
+				glog.Errorf("Failed to cleanup container status for %q: %v", id, err)
+			}
+		}
+	}()
+
 	// Add container into container store.
-	if err := c.containerStore.Create(meta); err != nil {
-		return nil, fmt.Errorf("failed to add container metadata %+v into store: %v",
-			meta, err)
+	if err := c.containerStore.Add(container); err != nil {
+		return nil, fmt.Errorf("failed to add container %q into store: %v", id, err)
```

Comment:
I see the storing of the status in the NewContainer() call but don't understand why the status is stored before the container. Is that to make sure the container always has status? maybe the container store.add could be changed to add container with status to make it more atomic?